torch/_decomp/decompositions.py

```diff
@@ -2860,7 +2860,7 @@ def nll_loss_forward(
     elif ignore_index >= 0:
         total_weight = (target != ignore_index).sum().to(self)
     else:
-        total_weight = self.new_full((), 1.0 * result.numel())
+        total_weight = self.new_ones(()) * 1.0 * result.numel()
```

Comment:
It would be much better to fix `new_full`. In this case `sympy.Expr` should be passed to `ops.index_expr` rather than `ops.constant` in the snippet you posted.

Reply:
Yeah it shouldn't be necessary to do this rewrite